src/core/gui/XournalView.cpp

```diff
@@ -553,16 +553,9 @@ void XournalView::zoomChanged() {
 
 void XournalView::pageSizeChanged(size_t page) {
     layoutPages();
-    if (page != npos && page < this->viewPages.size()) {
-        this->viewPages[page]->rerenderPage();
-    }
 }
 
-void XournalView::pageChanged(size_t page) {
-    if (page != npos && page < this->viewPages.size()) {
-        this->viewPages[page]->rerenderPage();
-    }
-}
```

Comment:
And here as well

Reply:
`XournalView::pageChanged` is called via `DocumentHandler::firePageChange`, which is used when the page background is changed (either the color `Control::changePageBackgroundColor()` or the pattern `PageBackgroundChangeController::commitPageTypeChange`), and in the timer callback mentionned in https://github.com/xournalpp/xournalpp/issues/4580#issuecomment-1381807460.

So no, it is useful, even if we remove the timer story.
It is also what you expect of this function: if anyone fires a PageChanged event, then the receiving XournalView should update said page. Doing nothing in such a callback function, while any dev would assume otherwise, is very much prone to errors.

Reply:
I disagree. Quick and dirty fixes are not a good idea in the long run.
I think the best course of action here is to restore those callback functions as they were, merge the rest of your changes (I'll rereview before that), wait for #4158 to get merged and then see if the rerender timer can be cleaned up.
It ain't a lot of work, but a bit of patience, I guess.